File: app/src/main/java/com/hazelmobile/filetransfer/ui/fragment/MusicListFragment.java
package com.hazelmobile.filetransfer.ui.fragment;

import android.content.Context;
import android.os.Bundle;
import android.provider.MediaStore;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.hazelmobile.filetransfer.R;
import com.hazelmobile.filetransfer.callback.Callback;
import com.hazelmobile.filetransfer.util.AppUtils;
import com.hazelmobile.filetransfer.app.EditableListAdapter;
import com.hazelmobile.filetransfer.app.EditableListFragment;
import com.hazelmobile.filetransfer.ui.adapter.MusicListAdapter;
import com.hazelmobile.filetransfer.ui.callback.TitleSupport;

import org.jetbrains.annotations.NotNull;

public class MusicListFragment
        extends EditableListFragment<MusicListAdapter.SongHolder, EditableListAdapter.EditableViewHolder, MusicListAdapter>
        implements TitleSupport {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setEmptyImage(R.drawable.ic_library_music_white_24dp);
        setEmptyText(getString(R.string.text_listEmptyMusic));
    }

    @Override
    public void onResume() {
        super.onResume();

        getContext().getContentResolver()
                .registerContentObserver(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, true, getDefaultContentObserver());
    }

    @Override
    public void onPause() {
        super.onPause();

        getContext()
                .getContentResolver()
                .unregisterContentObserver(getDefaultContentObserver());
    }

    @Override
    public MusicListAdapter onAdapter() {
        final AppUtils.QuickActions<EditableListAdapter.EditableViewHolder> quickActions = new AppUtils.QuickActions<EditableListAdapter.EditableViewHolder>() {
            @Override
            public void onQuickActions(final EditableListAdapter.EditableViewHolder clazz) {
                registerLayoutViewClicks(clazz);

                clazz.getView().findViewById(R.id.selector).setOnClickListener(
                        new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                if (getSelectionConnection() != null) {
                                    getSelectionConnection().setSelected(clazz.getAdapterPosition());
                                    Callback.setColor(true);
                                } else {
                                    Callback.setColor(false);
                                }
                            }
                        });
            }
        };

        return new MusicListAdapter(getActivity()) {
            @NonNull
            @Override
            public EditableViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
                return AppUtils.quickAction(super.onCreateViewHolder(parent, viewType), quickActions);
            }
        };
    }

    @Override
    public boolean onDefaultClickAction(EditableListAdapter.EditableViewHolder holder) {
        if (getSelectionConnection() != null) {
            Callback.setColor(true);
            return getSelectionConnection().setSelected(holder);
        } else {
            Callback.setColor(false);
            return performLayoutClickOpen(holder);
        }
    }

    @NotNull
    @Override
    public CharSequence getTitle(Context context) {
        return context.getString(R.string.text_music);
    }
}

File: app/src/main/java/com/hazelmobile/filetransfer/ui/fragment/MusicListFragment.kt
package com.hazelmobile.filetransfer.ui.fragment

import android.content.Context
import android.os.Bundle
import android.provider.MediaStore
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.genonbeta.android.framework.util.FileUtils

import com.hazelmobile.filetransfer.R
import com.hazelmobile.filetransfer.model.SongHolder
import com.hazelmobile.filetransfer.ui.adapter.MusicListAdapter
import com.hazelmobile.filetransfer.util.TimeUtils
import com.hazelmobile.filetransfer.util.callback.TitleSupport
import kotlinx.android.synthetic.main.music_list_fragment.*

class MusicListFragment : Fragment(), TitleSupport {

    override fun getTitle(context: Context): CharSequence {
        return "Audio"
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.music_list_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val audioList = ArrayList<SongHolder>()

        val songCursor = context?.contentResolver?.query(
            MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
            null,
            MediaStore.Audio.Media.IS_MUSIC + "=?",
            arrayOf(1.toString()), null
        )

        if (songCursor != null) {
            if (songCursor.moveToFirst()) {

                val nameIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.DISPLAY_NAME)
                val dateIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.DATE_MODIFIED)
                val sizeIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.SIZE)

                do {
                    val songSize = songCursor.getString(sizeIndex)
                    val songDate = songCursor.getLong(dateIndex)
                    val songHolder = SongHolder(
                        songCursor.getString(nameIndex),
                        FileUtils.sizeExpression(songSize.toLong(), false) + ", " +
                                TimeUtils.formatDateTime(context, songCursor.getLong(dateIndex) * 1000)
                    )
                    audioList.add(songHolder)
                } while (songCursor.moveToNext())

            }

            songCursor.close()
        }


        audioView.apply {
            audioView.layoutManager = LinearLayoutManager(context)
            val musiListAdapter = MusicListAdapter(audioList, context)
            audioView.adapter = musiListAdapter
        }

    }

}
